src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1976,6 +1978,22 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
         $transportRetryReferences = [];
         foreach ($config['transports'] as $name => $transport) {
             $serializerId = $transport['serializer'] ?? 'messenger.default_serializer';
+
+            if (isset($transport['symfony_serializer'])) {
+                $serializerDefinition = $container->findDefinition($serializerId);
+
+                if (!isset(class_implements($serializerDefinition->getClass())[FormatAndContextAwareSerializerInterface::class])) {
+                    throw new InvalidArgumentException(sprintf('Serializer for transport %s should implement %s in order to have custom format or context.', $name, FormatAndContextAwareSerializerInterface::class));
+                }
+
+                $container->setDefinition($serializerWithCustomConfigurationId = "messenger.transport.{$name}.serializer", new ChildDefinition($serializerId))
+                    ->addMethodCall('setFormat', [$transport['symfony_serializer']['format'] ?? $config['serializer']['symfony_serializer']['format']])
+                    ->addMethodCall('setContext', [($transport['symfony_serializer']['context'] ?? []) + $config['serializer']['symfony_serializer']['context']])
+                ;
```

Comment:
new definition(s) is needed here: if we want different config with the same service (for example default symfony's serializer) it has to be duplicated.